test/jdk/javax/swing/JComboBox/JComboBoxActionEvent.java

```diff
@@ -0,0 +1,91 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8290399
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Tests if AquaL&F fire actionevent if combobox menu is displayed.
+ * @run main/manual JComboBoxActionEvent
+ */
+
+import java.awt.FlowLayout;
+
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+public class JComboBoxActionEvent {
+    private static final String instructionsText = " Click the arrow to display the menu.\n" +
+         "While the menu is displayed, edit the text to create a new value.\n" + 
+         "Type return.\n" + 
+         "If a dialog appears with \"ActionCommand received\"\n" +
+         "press Pass, else Fail";
+
+    private static JFrame frame;
+
+    public static void createAndShowGUI() throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+
+            JComboBox<String> comboBox = new JComboBox<>(new String[]
+                    { "Apple", "Orange", "Pear" });
+            comboBox.setEditable(true);
+            comboBox.addActionListener(e -> {
+               System.out.println("Action Listener called: " + e.getActionCommand());
+               if (e.getActionCommand().contains("comboBoxEdited")) {
+                   JOptionPane.showMessageDialog(null, "ActionCommand received");
+               }
+            });
+
+            FlowLayout layout = new FlowLayout();
+            JPanel panel = new JPanel(layout);
+            panel.add(comboBox);
+            frame = new JFrame("Test Editable Combo Box");
+            frame.getContentPane().add(panel);
+            frame.setVisible(true);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+        });
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+
+        PassFailJFrame pfjFrame = new PassFailJFrame("JScrollPane "
+                + "Test Instructions", instructionsText, 5);
+
+        createAndShowGUI();
+
+        pfjFrame.awaitAndCheck();
+    }
+}
```

Comment:
Tried automating the test but it seems to call BasicComboBoxUI#focusLost when ENTER key is pressed, which calls actionPerformed thus defeating the purpose.


```

public class TestEditableComboBox
{
    private static JFrame fr;
    private static JComboBox<String> cb;
    private static Point loc;
    private static Dimension dim;
    private static volatile boolean actionDone = false;

    public TestEditableComboBox()
    {
        // Instructions:
        // Click the arrow to display the menu.
        // While the menu is displayed, edit the text to create a new value.
        // Type return.
        // The action listener should be invoked.

        // Using Aqua LAF, the return dismisses the menu but does invoke action listeners.
        // A second return is needed to invoke actions listeners.
        // This behavior is inconsistent with other LAFs and native NSComboBox.

//        if (true) {
//            try {
//                //UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
//                UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
//            } catch (Exception e) {
//                System.err.println(e.toString());
//            }
//        }

        fr = new JFrame("Test Editable Combo Box");
        String[] values = new String[] { "Apple", "Orange", "Pear" };
        cb = new JComboBox<>(values);
        cb.setEditable(true);
        cb.addActionListener(e -> {
            //actionDone = true;
                System.err.println("Action listener called: " + e.getActionCommand());
                });

        Container c = fr.getContentPane();
        c.setLayout(new FlowLayout());
        c.add(cb);
        fr.pack();
        //fr.setBounds(400, 100, 400, 80);
        fr.setLocationRelativeTo(null);
        fr.setVisible(true);
    }

    public static void main(String[] args) throws Exception {

        java.util.List<String> lafs = Arrays.stream(UIManager.getInstalledLookAndFeels())
                                  .map(UIManager.LookAndFeelInfo::getClassName)
                                  .collect(Collectors.toList());
        for (final String laf : lafs) {
            try {
                try {
                    UIManager.setLookAndFeel(laf);
                    System.out.println("Testing L&F: " + laf);
                } catch (UnsupportedLookAndFeelException e) {
                    System.out.println("Skipping unsupported L&F: " + laf);
                }
                SwingUtilities.invokeAndWait(TestEditableComboBox::new);
                SwingUtilities.invokeAndWait(() -> {
                    loc = cb.getLocationOnScreen();
                    dim = cb.getSize();
                });
                Robot robot = new Robot();
                robot.waitForIdle();
                robot.delay(1000);
                robot.mouseMove(loc.x + dim.width - 2, loc.y);
                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_A);
                robot.keyRelease(KeyEvent.VK_A);
                robot.waitForIdle();
                robot.delay(1000);
                robot.keyPress(KeyEvent.VK_ENTER);
                robot.keyRelease(KeyEvent.VK_ENTER);
                robot.waitForIdle();
                robot.delay(1000);
//                if (!actionDone) {
//                    throw new RuntimeException("ActionEvent not fired");
//                }
            } finally {
                SwingUtilities.invokeAndWait(() -> {
                    if (fr != null) {
                        fr.dispose();
                    }
                });
            }
        }
    }
}

```